docs/DeveloperGuide.md

```diff
@@ -283,31 +283,297 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `Address Book` and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case: UC01 - Add a new contact**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. User requests to add a new contact to the Address Book.
+2. Address Book adds contact to the contact list.
 
-    Use case ends.
+   Use case ends.
+
+**Extensions**
+
+* 1a. The given input is invalid.
+
+    * 1a1. Address Book shows an error message.
+
+      Use case ends.
+
+**Use case: UC02 - Update a contact**
+
+**MSS**
+
+1. User requests to update details for a person in the Address Book.
+2. Address Book updates the details of the contact.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The given input is invalid.
+
+    * 1a1. Address Book shows an error message.
+
+      Use case ends.
+
+* 1b. The given index is invalid.
+
+    * 1b1. Address Book shows an error message.
+
+      Use case ends.
+
+**Use case: UC03 - List all contacts in Address Book**
```

Comment:
Better to add an extension for the case whereby user tries to list an empty address book